src/applications/gi/components/AccordionItem.jsx

```diff
@@ -3,72 +3,99 @@ import React from 'react';
 import _ from 'lodash';
 import classNames from 'classnames';
 import recordEvent from 'platform/monitoring/record-event';
+import { createId } from '../utils/helpers';
 
 class AccordionItem extends React.Component {
+  static propTypes = {
+    expanded: PropTypes.bool.isRequired,
+    children: PropTypes.node.isRequired,
+    button: PropTypes.string.isRequired,
+  };
+
+  static defaultProps = {
+    expanded: true,
+    section: false,
+  };
+
   constructor(props) {
     super(props);
-    this.toggle = this.toggle.bind(this);
     this.state = {
       expanded: props.expanded,
-      button: props.button,
     };
     this.id = _.uniqueId('accordion-item-');
   }
 
-  toggle() {
-    this.setState({ expanded: !this.state.expanded });
+  componentDidUpdate(prevProps) {
+    if (prevProps.expanded !== this.props.expanded) {
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ expanded: this.props.expanded });
+    }
+  }
+
+  toggle = () => {
+    const expanded = !this.state.expanded;
+    this.setState({ expanded });
+
+    if (this.props.onClick) {
+      this.props.onClick(expanded);
+    }
```

Comment:
You could probably use [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#Optional_chaining_with_function_calls) here like so:

```js
this.props.onClick?.(expanded)
```